x-pack/plugins/security_solution/public/management/cypress/e2e/artifact_tabs_in_policy_details.cy.ts

```diff
@@ -0,0 +1,221 @@
+import { login, loginWithCustomRole, loginWithRole, ROLE } from '../tasks/login';
+import { performUserActions } from '../tasks/perform_user_actions';
+
+const loginWithPrivilegeAll = () => {
+  loginWithRole(ROLE.endpoint_security_policy_manager);
+};
+
+const loginWithPrivilegeRead = (privilegePrefix: string) => {
+  const roleWithArtifactReadPrivilege = getRoleWithArtifactReadPrivilege(privilegePrefix);
+  loginWithCustomRole('roleWithArtifactReadPrivilege', roleWithArtifactReadPrivilege);
+};
+
+const loginWithPrivilegeNone = (privilegePrefix: string) => {
+  const roleWithoutArtifactPrivilege = getRoleWithoutArtifactPrivilege(privilegePrefix);
+
+  loginWithCustomRole('roleWithoutArtifactPrivilege', roleWithoutArtifactPrivilege);
+};
+
+const getRoleWithArtifactReadPrivilege = (privilegePrefix: string) => {
+  const endpointSecurityPolicyManagerRole = getEndpointSecurityPolicyManager();
+
+  return {
+    ...endpointSecurityPolicyManagerRole,
+    kibana: [
+      {
+        ...endpointSecurityPolicyManagerRole.kibana[0],
+        feature: {
+          ...endpointSecurityPolicyManagerRole.kibana[0].feature,
+          siem: [
+            ...endpointSecurityPolicyManagerRole.kibana[0].feature.siem.filter(
+              (privilege) => privilege !== `${privilegePrefix}all`
+            ),
+            `${privilegePrefix}read`,
+          ],
+        },
+      },
+    ],
+  };
+};
+
+const getRoleWithoutArtifactPrivilege = (privilegePrefix: string) => {
+  const endpointSecurityPolicyManagerRole = getEndpointSecurityPolicyManager();
+
+  return {
+    ...endpointSecurityPolicyManagerRole,
+    kibana: [
+      {
+        ...endpointSecurityPolicyManagerRole.kibana[0],
+        feature: {
+          ...endpointSecurityPolicyManagerRole.kibana[0].feature,
+          siem: endpointSecurityPolicyManagerRole.kibana[0].feature.siem.filter(
+            (privilege) => privilege !== `${privilegePrefix}all`
+          ),
+        },
+      },
+    ],
+  };
+};
+
+const visitArtifactTab = (tabId: string) => {
+  visitPolicyDetailsPage();
+  cy.get(`#${tabId}`).click();
+};
+
+const visitPolicyDetailsPage = () => {
+  cy.visit('/app/security/administration/policy');
+  cy.getBySel('policyNameCellLink').eq(0).click({ force: true });
+  cy.getBySel('policyDetailsPage').should('exist');
+};
+
+describe('Artifact tabs in Policy Details page', () => {
+  before(() => {
+    login();
+    loadEndpointDataForEventFiltersIfNeeded();
+    removeAllArtifacts();
+  });
+
+  after(() => {
+    removeAllArtifacts();
+  });
+
+  for (const testData of getArtifactsListTestsData()) {
+    describe(`${testData.title} tab`, () => {
+      it(`[NONE] User cannot see the tab for ${testData.title}`, () => {
+        loginWithPrivilegeNone(testData.privilegePrefix);
+        visitPolicyDetailsPage();
+
+        cy.get(`#${testData.tabId}`).should('not.exist');
+      });
+
+      context(`Given there are no ${testData.title} entries`, () => {
```

Comment:
I like to differentiate the functionality under test using `describe`, and the `context` of the grouped test cases - but this is only a style preference. do we try to avoid using `context`?

Reply:
No, we don't really avoid `context`. I just had to look it up in the mocha docs to see what the difference was w.r.to `describe`, and so I wondered if it is obvious for everyone who reads the test file to grasp `context` right off the bat or would have to look it up like me. 😅

Reply:
which is why is suggested to use `describe` that we usually see in test files.

Reply:
I see - in this case I'd keep the `context` :)